amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/activity/validators/TreeCollectionValidator.java

```diff
@@ -147,7 +147,7 @@ private boolean isPresentParentProgramInCollection(Long programId, Set<Long> idV
                 } 
             }
         } catch (DgException e) {
-            throw new RuntimeException(e);
+            logger.error(e.getMessage(), e);
```

Comment:
In fact if we expect "valid" input here, then throwing a `RuntimeException` is right to flag that something is wrong (like in this case we had to run business rule check after subelements validation). Please restore.

Reply:
I've kept logging the message. It was hard to understand the root of the problem without stack trace.

Reply:
This is fine, logging more doesn't hurt :) but good that you restored the `RuntimeException`.